### ISSUE TITLE
setBadgeBackgroundColor error

### ISSUE THREAD

**Issue by code4you2021** [wxt-dev/wxt#179]
### Bug Report

<img width="563" alt="" src="https://github.com/wxt-dev/wxt/assets/83153991/da3ee717-a6cd-4298-aa17-a736bd503b74">

### Bug Test
I wrote three demos(`without-wxt`, `wxt-with-popup`, `wxt-without-popup`) to test the following code.
https://github.com/code4you2021/wxt-demo/tree/main/demo-001

`wxt` version: `^0.8.0`

background.js
```javascript

chrome.action.setBadgeBackgroundColor({ color: "#FF0000" });
chrome.action.setBadgeText({ text: "100" });
chrome.action.setBadgeTextColor({ color: "#FFFFFF" });

browser.action.setBadgeBackgroundColor({ color: "#FF0000" });
browser.action.setBadgeText({ text: "100" });
browser.action.setBadgeTextColor({ color: "#FFFFFF" });
```

The result:
chrome.action.setBadgeXXX does not report an error: ✅ 
chrome.action.setBadgeXXX report an error: ❌ 

|Demo | Result |
| ------| -------| 
| [without-wxt](https://github.com/code4you2021/wxt-demo/tree/main/demo-001/without-wxt) | ✅ |
| [wxt-with-popup](https://github.com/code4you2021/wxt-demo/tree/main/demo-001/wxt-with-popup) | ✅ |
| [wxt-without-popup](https://github.com/code4you2021/wxt-demo/tree/main/demo-001/wxt-without-popup)| ❌ |

### Reasons For Bugs

the `wxt-without-popup` manifest.json doesn't have the `action` field

![](https://github.com/wxt-dev/wxt/assets/83153991/0e7ded81-5eca-42f8-a7f4-6af389e2ee61)

### Try To Solve

Tried to add the missing field in `manifest.json`, but vscode reported an error.

<img width="730" alt="" src="https://github.com/wxt-dev/wxt/assets/83153991/67726222-de17-4090-ad3f-4e4e5fd1817c">
<img width="1095" alt="" src="https://github.com/wxt-dev/wxt/assets/83153991/8d9579ed-8039-47f4-b0ff-ea52e0937262">

### How To Solve

Personal advice, for reference only.

When there is no popup page, `manifest.json` adds the action field by default.

```json
{
    "action": {
        "default_icon": {
          "128": "icon/128.png"
        },
        "default_title": "without-wxt" // get the title from the name field
    },
}
```

**Comment by aklinker1** [comment]
I always forget you can have an action without a popup. I'll fix this tonight 👍

**Comment by aklinker1** [comment]
Released in [`v0.8.4`](https://github.com/wxt-dev/wxt/releases/tag/v0.8.4)

Add the action to the manifest.json like you did in this image, there shouldn't be a type error and the final manifest will include an action without a popup.

![image](https://github.com/wxt-dev/wxt/assets/10101283/5be9787f-d0cc-44f0-bb4e-90caad433e04)
